Articles/Blog/HelloHeatEquation.md

```diff
@@ -0,0 +1,226 @@
+# The Heat Equation is the "Hello World" of Scientific Computing
+
+#### Contributed by [Heather M. Switzer](https://github.com/heatherms27 "Heather Switzer GitHub Profile") and [Elsa Gonsiorowski](https://github.com/gonsie "Elsa Gonsiorowski GitHub Profile")
+
+"Hello World!" is a prototypical program that is universally used by programmers to illustrate the basic syntax of a programming language. At a glance, it can demonstrate a programming language's syntactic structure and can indicate how easy or difficult implementing a basic program may be (see several [Hello World implementations at WikiBooks](https://en.wikibooks.org/wiki/Computer_Programming/Hello_world)). For the scientific software community, a programming language is only the most basic building block of a program. Scientific software must use various programming models or math libraries to achieve high performance for any particular problem. These communities need a version "Hello World!" to compare programming options. The Heat Equation, a well known partial differential equation (PDE), is the perfect formula for this task.
+
+## What is the Heat Equation?
+The Heat Equation, also known as the Diffusion Equation, was developed by Joseph Fourier in 1822. Primarily used in the fields of Mathematics and Physics, the equation shows how heat (or something similar) changes over time in a solid medium. While there are many adaptations for different dimensions and scenerios, new programmers are often first introduced to the 1-dimensional version of the Heat Equation as a tutorial problem. Specifically, the Forward-Time, Central-Space (FTCS) version of the equation is used.
+
+The following description is adapted from [a Trinity University lecture](http://ramanujan.math.trinity.edu/rdaileda/teach/s14/m3357/lectures/lecture_2_25_slides.pdf "One-Dimensional Heat Equation").
+
+Suppose you have a thin metal rod along the x-axis starting at position 0 and ending at position _L_. The goal is to model the temperature distribution throughout the rod at a given time, with the following (ideal) assumptions:
+
+* An initial temperature distribution is given
+* Boundary conditions (established tempuratures at each end of the rod) are known
```

Comment:
I am curious. I know the rod *feels* more one dimensional but I wonder if that is better than a *wall* which we consider infinite in extent and so focus only on the dimension *through* the wall.

Reply:
I really struggled with the wall image that was in your original heat-equation example. The rod makes more sense to me. I can visualize the heat change along the rod... A wall in my mind is a 2D object.

Reply:
I get that, I guess. I just wonder if your experience is rather unique or rather common. I also wonder if there some key explanation missing that would help avoid that struggle for new readers. The reason the wall IMHO is a better model is that it affords some simple extentsions of the problem for more interesting scenarios and helps to define a sciencey problem that is more true-to-life and real-world experience than a simple rod.

Reply:
I agree with both statements. In my mind for a simple 1D problem, the rod is easier to visualize with heat just crossing through one axis. The wall example does seem more practical and easier to extend, but it also reminds me of the heated plate example, which seems to be a common when describing the 2D version as heat travels through a mesh structure along the x- and y-axes. (Seen here: http://ramanujan.math.trinity.edu/rdaileda/teach/s12/m3357/lectures/lecture_3_6_short.pdf). My reasoning was that because this article was more focused on the software side of the coin, the rod would be easier to understand, while if it was a more math-centered blog post the wall may be better. That's just my 2 cents though.